File: spark/src/main/scala/com/jason/spark/core/Spark14_Action_CountByKey.scala
package com.jason.spark.core

import org.apache.spark.{SparkConf, SparkContext}

object Spark14_Action_CountByKey {
  def main(args: Array[String]): Unit = {
    val sparkConf = new SparkConf().setMaster("local[*]").setAppName("Spark11_KeyValue_Operation")
    val sc = new SparkContext(sparkConf)
    val dataRDD = sc.makeRDD(List(("a", 8),("a", 8),("b", 9),("c", 10)))
    val stringToLong = dataRDD.countByKey()
    stringToLong.foreach(println)
  }

}

File: spark/src/main/java/com/jason/spark/core/rddtest/Spark14ActionCountByKey.java
package com.jason.spark.core.rddtest;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaSparkContext;
import scala.Tuple2;

import java.util.Arrays;
import java.util.Map;

/**
 * @Description:
 * @author: 贾森
 * @date: 2024年10月28日 15:17
 */
public class Spark14ActionCountByKey {
    public static void main(String[] args) {
        // 1.创建配置对象
        SparkConf conf = new SparkConf().setMaster("local[*]").setAppName("sparkCore");

        // 2. 创建sparkContext
        JavaSparkContext sc = new JavaSparkContext(conf);

        // 3. 编写代码
        JavaPairRDD<String, Integer> pairRDD = sc.parallelizePairs(Arrays.asList(new Tuple2<>("a", 8), new Tuple2<>("b", 8), new Tuple2<>("a", 8), new Tuple2<>("d", 8)));
        Map<String, Long> stringLongMap = pairRDD.countByKey();
        for (String s : stringLongMap.keySet()){
            System.out.println(s+"   "+stringLongMap.get(s));
        }
    }
}
